Provide justification document for unresolved messages (#127)
thoth-station/thoth-station.github.io: _j/unresolved.md

---
title: Resolver failed as it was unable to resolve direct dependencies
---

The resolution was terminated as direct dependencies could not be resolved.

## Issue description

> To resolve a software stack, Thoth aggergates information packages published on

Reviewer: ```suggestion
To resolve a software stack, Thoth aggregates information packages published on
```

Python package indexes. This message happens when Thoth has not solved a direct

> dependency a user's stack requires. As the direct dependency was not analyzed

Reviewer: ```suggestion
dependency required by user's stack. As the direct dependency was not analyzed
```

yet, Thoth cannot resolve the dependency graph to deliver a pinned down
software stack.

> On the backend side, this prioritizes analyses of the missing package so that

Reviewer: ```suggestion
On the backend side, this prioritizes analysis of the missing package so that
```

users can receive recommendations as soon as possible.

## Affected packages:

> This message is not specific to any concrete package.

Reviewer: ```suggestion
This message is not specific to any specific package.
```

## Severity

 * ERROR

## Issue fix

Get back to Thoth in a while. Thoth backend analyzes the software to provide
recommendations.

> Note this can happen also when the package requested is not available on PyPI

Reviewer: ```suggestion
Note that this can happen also when the requested package is not available on PyPI
```
